src/components/Account/AccountProfileContainer.tsx

```diff
@@ -0,0 +1,255 @@
+import * as Arc from "@daostack/arc.js";
+import promisify = require("es6-promisify");
+import * as ethUtil from 'ethereumjs-util';
+import * as sigUtil from 'eth-sig-util';
+import { Formik, Field, FormikBag } from 'formik';
+import * as queryString from 'query-string';
+import * as React from "react";
+import { connect, Dispatch } from "react-redux";
+import { Link, RouteComponentProps } from "react-router-dom";
+
+import * as profileActions from "actions/profilesActions";
+import { IRootState } from "reducers";
+import { IAccountState, IDaoState } from "reducers/arcReducer";
+import { NotificationStatus, showNotification } from "reducers/notifications";
+import { IProfileState } from "reducers/profilesReducer";
+import Util from "lib/util";
+
+import AccountImage from "components/Account/AccountImage";
+import ReputationView from "components/Account/ReputationView";
+
+import * as css from "./Account.scss";
+
+interface IStateProps extends RouteComponentProps<any> {
+  accountAddress: string;
+  accountInfo?: IAccountState;
+  accountProfile?: IProfileState;
+  currentAccountAddress: string;
+  dao?: IDaoState;
+}
+
+const mapStateToProps = (state: IRootState, ownProps: any) => {
+  const queryValues = queryString.parse(ownProps.location.search);
+
+  return {
+    accountAddress: ownProps.match.params.accountAddress,
+    accountInfo: state.arc.accounts[state.web3.ethAccountAddress + "-" + queryValues.daoAvatarAddress],
+    accountProfile: state.profiles[state.web3.ethAccountAddress],
+    currentAccountAddress: state.web3.ethAccountAddress,
+    dao: queryValues.daoAvatarAddress ? state.arc.daos[queryValues.daoAvatarAddress as string] : null
+  };
+};
+
+interface IDispatchProps {
+  showNotification: typeof showNotification;
+  updateProfile: typeof profileActions.updateProfile;
+}
+
+const mapDispatchToProps = {
+  showNotification,
+  updateProfile: profileActions.updateProfile
+};
+
+type IProps = IStateProps & IDispatchProps;
+
+interface IState {
+  genCount: number;
+  ethCount: number;
+}
+
+interface FormValues {
+  description: string;
+  githubURL: string;
+  name: string;
+}
+
+class AccountProfileContainer extends React.Component<IProps, IState> {
+
+  constructor(props: IProps) {
+    super(props);
+
+    this.state = {
+      ethCount: null,
+      genCount: null
+    };
+  }
+
+  public async componentWillMount() {
+    const { accountAddress, dao } = this.props;
+    const web3 = await Arc.Utils.getWeb3();
+    const getBalance = promisify(web3.eth.getBalance);
+    const ethBalance = await getBalance(accountAddress);
+
+    let votingMachineInstance: Arc.GenesisProtocolWrapper;
+    if (dao) {
+      const contributionRewardInstance = await Arc.ContributionRewardFactory.deployed();
```

Comment:
This is a lot of logic just to get the GEN balance. arc.js does calls to the blockchain for each of these `deployed` and `at` calls. Wouldn't it make more sense to have all these addressed in the redux state, so they are easily re-used?

Reply:
Arc.js doesn't go to the blockchain for the `deployed` calls.  It gets the addresses and code from the contract json files.  But it does for `at` calls.

Reply:
Further, Arc.js caches everything associated with getting the hydrated wrapper instance.  So as far as the Arc.js part of this is concerned, it is quite efficient.

Reply:
Ok, perhaps i misunderstand. 
It seems to be that you are using the truffle-contract `deployed` and `at` functions, which are doing a network check by polling the blockchain:  https://github.com/trufflesuite/truffle/blob/next/packages/truffle-contract/lib/contract.js#L231,  and for the at() function here: https://github.com/trufflesuite/truffle/blob/next/packages/truffle-contract/lib/contract.js#L231

Reply:
@jellegerbrandy Ah, you are right it is checking for the ability to connect to the network (rather inefficiently I must say since it caches the result and could be just checking for the cached networkId.

In any case, after the first call Arc.js caches the result and ensuing calls will not access the net.